File: python-euler/utils/BinaryTrees/BTNode.py
class BTNode(object):
	"""
	Node object for basic binary tree structure.
	"""

	def __init__(self, data = None, parent = None):
		"""
		Implementing the tree where data can be an object.
		Should probably refactor to use key/value pairs
		for nodes.

		@return: node object
		"""
		self.data = data
		self.left = None
		self.right = None
		self.parent = parent

	def __str__(self):
		"""
		@return: string containing the node's data
		"""
		return str(self.data)

	def children_count(self):
		"""
		@return: number of children: 0, 1, 2
		"""
		count = 0
		if self.left:
			count += 1
		if self.right:
			count += 1
		return count

File: java-euler/src/utils/BTNode.java
package utils;

public class BTNode {
	public BTNode left, right, parent;
	public int data;
	
	public BTNode(int data) {
		this.data = data;
		this.left = null;
		this.right = null;
		this.parent = null;
	}
	
	public BTNode(int data, BTNode parent) {
		this.data = data;
		this.left = null;
		this.right = null;
		this.parent = parent;
	}
	
	public int children_count() {
		int count = 0;
		if (this.left != null) count++;
		if (this.right != null) count++;
		return count;
	}

	@Override
	public String toString() {
		return String.format("%d", this.data);
	}
}
